lib/transaction.cc

```diff
@@ -1265,7 +1265,8 @@ static int verifyPackage(rpmts ts, rpmte p, struct rpmvs_s *vs, int vfylevel)
     if (fd != NULL) {
 	ARGI_t ids = initPkgDigests(fd);
 	prc = rpmpkgRead(vs, fd, NULL, NULL, &vd.msg);
-	finiPkgDigests(fd, ids, prc ? NULL : auxh);
+	int test = rpmtsFlags(ts) & RPMTRANS_FLAG_TEST;
+	finiPkgDigests(fd, ids, (test || prc) ? NULL : auxh);
```

Comment:
Pretty sure you've thought about it but I'll ask anyway (if just to understand): Why don't we instead check that the package digest (for the given algorithm) isn't yet present in the aux header?

Reply:
Just because doing that is way more complicated. Yep, I did think about it.

Reply:
Yep, that was the no. 1 explanation in my mind, thanks.